hil/api.py

```diff
@@ -690,18 +690,35 @@ def headnode_detach_network(headnode, hnic):
 @rest_call('GET', '/networks', Schema({}))
 def list_networks():
     """Lists all networks"""
-
-    get_auth_backend().require_admin()
-
-    networks = db.session.query(model.Network).all()
-    result = {}
-    for n in networks:
-        if n.access:
-            net = {'network_id': n.network_id,
-                   'projects': sorted([p.label for p in n.access])}
-        else:
-            net = {'network_id': n.network_id, 'projects': None}
-        result[n.label] = net
+    # Admin Operation
+    if get_auth_backend().have_admin():
+        networks = db.session.query(model.Network).all()
+        result = {}
+        for n in networks:
+            if n.access:
+                net = {'network_id': n.network_id,
+                       'projects': sorted([p.label for p in n.access])}
+            else:
+                net = {'network_id': n.network_id, 'projects': None}
+            result[n.label] = net
+    # Regular User Operation
+    else:
+        networks = db.session.query(model.Network).all()
```

Comment:
You coud do this before your if else starts, so you don't have to write it twice. Same thing with `result ={} `

Reply:
Great!